### PR TITLE
New fingerprints for OS X

### PR DESCRIPTION
**Pull request by jkennedy-r7**
* Added banner fingerprint for 10.3 using QuickTime Stream Server.
* Added banner fingerprint for 10.5 and 10.6 using Samba

### CODE REVIEW

File: xml/http_servers.xml [hunk 1]
```diff
@@ -3510,4 +3510,16 @@
       <param pos="0" name="service.product" value="gSOAP"/>
       <param pos="1" name="service.version"/>
     </fingerprint>
+
+           <!-- Apple QuickTime streaming server -->
+    <fingerprint pattern="Platform\/MacOSX; Release\/Panther;">
```
Comment (jhart-r7) [inline xml/http_servers.xml]: Why not anchor this and also fingerprint the service.

**Comment by gwiseman-r7** [comment]
:anchor: ship it

File: xml/http_servers.xml [hunk 2]
```diff
@@ -3510,4 +3510,29 @@
...
+      <param pos="0" name="os.vendor" value="Apple"/>
+      <param pos="0" name="os.family" value="Mac OS X"/>
+      <param pos="0" name="os.product" value="Mac OS X"/>
+      <param pos="0" name="os.device" value="General"/>
+      <param pos="0" name="os.version" value="10.3"/>
+      <param pos="0" name="service.product" value="QTSS"/>
+      <param pos="1" name="service.version"/>
+    </fingerprint>
+
+    <fingerprint pattern="^QTSS\/([\d*.]*) \(Build\/[\d*.]*; Platform\/MacOSX; Release\/Mac OS X">
```
Comment (jhart-r7) [inline xml/http_servers.xml]: Here and the other places where you are matching version-like strings, I think `[\d*.]*` could be improved -- the `*` inside will match the literal *, the `.` should be explicitly escaped and the  `*` at the end should probably be a `+`.  I'd suggest `[\d\.]+`.

File: xml/smb_native_os.xml [hunk 3]
```diff
@@ -384,6 +384,33 @@
       <param pos="2" name="os.build"/>
    </fingerprint>
 
+      <!-- Mac OS X -->
+   <fingerprint pattern="^Samba 3.0.28a-apple">
+      <description>OS X 10.6</description>
```
Comment (jhart-r7) [inline xml/smb_native_os.xml]: Perhaps mention Samba in here

File: xml/smb_native_os.xml [hunk 4]
```diff
@@ -384,6 +384,33 @@
...
+      <param pos="0" name="os.family" value="Mac OS X"/>
+      <param pos="0" name="os.product" value="Mac OS X"/>
+      <param pos="0" name="os.device" value="General"/>
+      <param pos="0" name="os.version" value="10.6"/>
+      <param pos="0" name="service.vendor" value="Samba"/>
+      <param pos="0" name="service.product" value="Samba"/>
+      <param pos="0" name="service.version" value="3.0.28a-apple"/>
+   </fingerprint>
+
+   <fingerprint pattern="^Samba 3.0.25b-apple">
```
Comment (jhart-r7) [inline xml/smb_native_os.xml]: Escape those periods and add a trailing anchor.  Also, even though you are looking for an explicit version, IMO it is simpler, cleaner and easier to add more in the future if you capture the version up here and use it in the `param` down below with a `pos` of 1.

File: xml/http_servers.xml (hunk 2 above)
Comment (jkennedy-r7) [inline xml/http_servers.xml]: Fixed

File: xml/smb_native_os.xml (hunk 3 above)
Comment (jkennedy-r7) [inline xml/smb_native_os.xml]: Added

File: xml/smb_native_os.xml (hunk 4 above)
Comment (jkennedy-r7) [inline xml/smb_native_os.xml]: Addressed

File: xml/smb_native_os.xml [hunk 5]
```diff
@@ -384,6 +384,33 @@
       <param pos="2" name="os.build"/>
    </fingerprint>
 
+      <!-- Mac OS X -->
+   <fingerprint pattern="^Samba (3\.0\.28a-apple)$">
```
Comment (jhart-r7) [inline xml/smb_native_os.xml]: Better, but I'm not 100% sure we want to be including the `-apple` part in the version.  There are pros and cons to doing it and not doing it.  I don't have a strong opinion either way but it is worth thinking about for the future.

Comment (jkennedy-r7) [inline xml/smb_native_os.xml]: I can get both the point of including it or not. I guess we might want the version recored in the same way authenticated version check is done. If you run `smbd --version`,  the version string returned contains the `-apple`.

Comment (jhart-r7) [inline xml/smb_native_os.xml]: The `smbd --version` point is proof enough.  Lets keep it as is.

To the point about authenticated version checks, keep in mind that recog is a generic framework that is not specific to Nexpose, metasploit or even necessarily any Rapid7 product (although today they are the primary consumers).  We need to ensure that any changes we make here are not overloaded with things that are product specific.

Comment (mdobrska-r7) [inline xml/smb_native_os.xml]: :shipit:

**Pull request merged**